File: Kode/Watch/Android/BrregWear/app/src/main/java/com/bronnoysund/wear/ui/CompanyDataRow.kt
// SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Commercial

package com.bronnoysund.wear.ui

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Divider
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.wear.compose.material.MaterialTheme
import androidx.wear.compose.material.Text

@Composable
fun CompanyDataRow(label: String, value: String) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp),
    ) {
        Text(
            text = label,
            style = MaterialTheme.typography.caption1,
            color = MaterialTheme.colors.onSurfaceVariant,
        )
        Divider(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 2.dp),
            thickness = 0.5.dp,
            color = MaterialTheme.colors.onSurface.copy(alpha = 0.3f),
        )
        Text(
            text = value,
            style = MaterialTheme.typography.body1,
            color = MaterialTheme.colors.onSurface,
        )
    }
}

File: Kode/Watch/iOS/BrregWatch/BrregWatch Watch App/CompanyDataRow.swift
// SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Commercial

import SwiftUI

struct CompanyDataRow: View {
    let label: String
    let value: String

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(label)
                .font(.caption2)
                .foregroundColor(.secondary)
            Divider()
                .opacity(0.3)
            Text(value)
                .font(.body)
                .foregroundColor(.primary)
        }
        .padding(.vertical, 4)
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}
